com.unity.visualeffectgraph/Shaders/Sort.compute

```diff
@@ -290,15 +310,14 @@ void MERGE_PASS(uint3 groupId           : SV_GroupID,
     } while (!done);
 
     // Select left or right: Must handle special case for equality
-    bool select0 = (key0 == key1 && (id & 1)) || (CompareKeys(key0,key1) != reverse);
-    uint value = inputSequence[(select0 ? index0 : index1)].value;
+    bool select0 = (kvp0.key == kvp1.key && kvp0.value == kvp1.value && (id & 1)) || (CompareKVP(kvp0, kvp1) != reverse);
 
 #if FINAL_PASS
-    sortedSequence[id] = value;
+    sortedSequence[id] = select0 ? kvp0.value : kvp1.value;
 #else
-    float key = select0 ? key0 : key1;
-    KVP kvp = { key, value };
-    sortedSequence[id] = kvp;
+    if (select0)
+        kvp1 = kvp0;
+    sortedSequence[id] = kvp1;
```

Comment:
I would use a ternary operator here

Reply:
It turns out that the compiler doesn't like user defined types in ternary operator, that's why it was like that in the first place :(